content/security/for-admins/scim.md

```diff
@@ -58,3 +61,189 @@ You must make sure you have [configured SSO](single-sign-on/configure/_index.md)
+6. After creating the attributes, go to the top and select **Mappings > Okta User to YOUR APP**.
+7. Go to the newly created attributes and map the variable names selected above to the external names, then select **Save Mappings**. If you’re using JIT provisioning, continue to the following step.
+8. Go to **Applications > YOUR APP > General > SAML Settings > Edit > Step 2** and configure the mapping from the user attribute to the docker variables.
+
+### Assign roles by user
+
+1. Go to **Directory > People > YOUR USER > Profile**, then select **Edit** on **Attributes**.
+2. Update the attributes to the desired values.
+
+### Assign roles by group
+
+1. Go to **Directory > People > YOUR GROUP > Applications > YOUR APPLICATION**, then select the **Edit** icon.
+2. Update the attributes to the desired values.
+
+If a user doesn't already have attributes set up, users who are added to the group will inherit these attributes upon provsioning.
+
+{{< /tab >}}
+{{< tab name="Entra ID SAML 2.0" >}}
+
+### Set up
+
+1. Setup [SSO](./single-sign-on/configure/_index.md) and SCIM first.
+2. In the Azure AD admin portal, go to**Enterprise Apps > YOUR APP > Provisioning > Mappings > Provision Azure Active Directory Users**.
+3. To set up the new mapping, check **Show advanced options**, then select **Edit attribute options**.
+4. Create new entries with the desired mapping for role, org, or group (for example, `urn:ietf:params:scim:schemas:extension:docker:2.0:User:dockerRole`) as a string type.
+5. Go back to **Attribute Mapping** for users and click **Add new mapping**.
+
+### Expression mapping
+
+This implementation works best for roles, but can't be used along with organization and team mapping using the same method. With this approach, you can assign attributes at a group level, which members can inherit. This is the recommended approach for role mapping.
+
+1. In the **Edit Attribute** view, select the **Expression** mapping type.
+2. To create app roles named as the role directly, in the **Expression** field, enter `SingleAppRoleAssignment([appRoleAssignments])`.
+3. Set the following fields:
+    - **Target attribute**: `urn:ietf:params:scim:schemas:extension:docker:2.0:User:dockerRole`.
+    - **Match objects using this attribute**: No
+    - **Apply this mapping**: Always
+4. Save your configuration.
+
+If you’re restricted to using app roles you have already defined (for example, `My Corp Administrators`) you’ll need to setup a switch for these roles. For example:
+
+```text
+Switch(SingleAppRoleAssignment([appRoleAssignments]), "member", "My Corp Administrator", "owner", "My Corp Editor", "editor")`
+```
+
+### Direct mapping
```

Comment:
suggestion: I think we should mention here that this requires the admin to assign users _only_ via the Microsoft Graph API instead of just using the Entra ID portal. This is a strong drawback of this approach, and without mentioning it, Direct Mapping probably would seem easier to setup to anyone reading the docs.

Reply:
Adding:

> In order to assign users, you'll need to use the Microsoft Graph API.